linux-tools/test_time/test_time.py

```diff
@@ -0,0 +1,200 @@
+                    self.output_file) and utils.file_contains_pattern(
+                    self.output_file, cmd_invoked):
+                logging.info(
+                    "\n Output file is created successfully using -o option of time utility")
+            else:
+                self.nfail += 1
+                raise error.TestError(
+                    "\n Failed to create output file using -o option")
+
+    def test_append_option(self):
+        """
+        This method will verify the --append option of time utility
+        Ex:/usr/bin/time -v -a -o time_out.txt <cmd>
+        """
+        cmd_invoked = "pwd"
+        cmd = "%s -v -a -o %s %s" % (self.time_exe,
+                                     self.output_file, cmd_invoked)
+        if (os.system(cmd) != 0):
+            self.nfail += 1
+            raise error.TestError(
+                '\n time command with -a option is failed to execute')
+        else:
+            if os.path.exists(
+                    self.output_file) and utils.file_contains_pattern(
+                    self.output_file, cmd_invoked):
+                logging.info(
+                    "\n Output is appended to the output file successfully" +
+                    "using -a option of time utility")
+            else:
+                self.nfail += 1
+                raise error.TestError(
+                    "\n Failed to append the output to a file using -a option")
+
+    def test_format_option(self):
+        """
+        This method will verify the --format option of time utility
+        Ex:/usr/bin/time -f"<format>" <cmd>
+        """
+        cmd_invoked = "sleep %s" % self.seconds
+        cmd = "%s " % self.time_exe + \
+            "-f \"%E real,\t%U user,\t%S system,\t%x status\" " + cmd_invoked
+        cmd_output = commands.getoutput(cmd)
+        if re.search(
+                "%s.00 real,\t\S+ user,\t\S+ system,\t\d+ status" %
+                self.seconds,
+                cmd_output):
+            logging.info(
+                "\n --format option of time utility is verified succesfully")
+        else:
+            self.nfail += 1
+            raise error.TestError(
+                "\n Failed to verify --format option of time utility")
+
+    def test_custom_script(self):
+        """
+        This method will test the custom shell script with the time utility
+        create a custom shell script and provide it as input to the time
+        Ex: /usr/bin/time -v <custom-script>
+        """
+        custom_script = "%s/custom_script.sh" % (self.tmpdir)
+        f = open(custom_script, 'w')
+        f.writelines("""
+#!/bin/bash
+echo "Hello, $LOGNAME"
+echo "Current date is `date`"
+echo "User is `who i am`"
+echo "Current direcotry `pwd`"
+""")
+        f.close()
+        cmd = "%s -v bash %s" % (self.time_exe, custom_script)
+        cmd_output = commands.getoutput(cmd)
+        if re.search("%s" % (custom_script), cmd_output):
+            logging.info(
```

Comment:
Here its better to check for the script output to verify execution of custom script than using script name in the output